EIPS/eip-eth_simulateTransaction.md

```diff
@@ -0,0 +1,109 @@
+
+This is terrible for user experience as it encourages users to be accustomed to signing transactions blindly, which may have adverse consequences such as loss of funds.
+
+A new JSONRPC method `eth_simulateTransaction` aims to improve the situation by providing a way for wallets to perform a dry run and read events emitted (such as the ERC20 `Transfer` events), thereby enabling the user to make a more informed decision about whether to sign a transaction that is requested.
+
+## Specification
+
+#### eth_simulateTransaction
+
+Performs a dry-run of the transaction and returns a transaction result object. The transaction will not be added to the blockchain. Note that the actual transaction may have a different result for a variety of reasons including EVM mechanics, node performance, and the difference in the state of the blockchain when the transaction is processed.
+
+##### Parameters
+
+1. `Object` - The transaction object
+  - `from`: `DATA`, 20 Bytes - The address the transaction is send from.
+  - `to`: `DATA`, 20 Bytes - (optional when creating new contract) The address the transaction is directed to.
+  - `gas`: `QUANTITY`  - (optional, default: 90000) Integer of the gas provided for the transaction execution. It will return unused gas.
+  - `gasPrice`: `QUANTITY`  - (optional, default: To-Be-Determined) Integer of the gasPrice used for each paid gas
+  - `value`: `QUANTITY`  - (optional) Integer of the value sent with this transaction
+  - `data`: `DATA`  - The compiled code of a contract OR the hash of the invoked method signature and encoded parameters. For details see [Ethereum Contract ABI](https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI)
+  - `nonce`: `QUANTITY`  - (optional) Integer of a nonce. This allows to overwrite your own pending transactions that use the same nonce.
+
+##### Example Parameters
+```js
+params: [{
+  "from": "0x1111111111111111111111111111111111111111",
+  "to": "0x2222222222222222222222222222222222222222",
+  "gas": "0x200000",
+  "gasPrice": "0x3b9aca00",
+  "value": "0x0",
+  "data": "0xdeadbeef000000000000000000000001111111111111111111111111111111111111110000000000000000000000000000000000000000000000000000000000abcde"
+}]
+```
+
+##### Returns
+
+`Object` - A transaction result object.
+
+  - `transactionHash`: `DATA`, 32 Bytes - hash of the transaction.
+  - `from`: `DATA`, 20 Bytes - address of the sender.
+  - `to`: `DATA`, 20 Bytes - address of the receiver. null when it's a contract creation transaction.
+  - `gasUsed`: `QUANTITY` - The amount of gas used by this specific transaction alone.
+  - `contractAddress`: `DATA`, 20 Bytes - The contract address created, if the transaction was a contract creation, otherwise `null`.
+  - `logs`: `Array` - Array of log objects, which this transaction generated.
```

Comment:
It feels like this should return the `status` as well.  Also, including the revert reason would be valuable if a revert occurred.